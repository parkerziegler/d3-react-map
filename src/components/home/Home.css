@import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");
.home-text {
  font: 24px "Source Code Pro", monospace; }
