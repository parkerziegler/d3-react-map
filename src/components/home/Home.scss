@import url('https://fonts.googleapis.com/css?family=Source+Code+Pro');

$sourcecode: 'Source Code Pro', monospace;

.home-text {
    font: 24px $sourcecode;
}