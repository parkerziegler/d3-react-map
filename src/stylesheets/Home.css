@import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");
/** A constants SCSS partial for storing common variables and mixins */
@import url("https://fonts.googleapis.com/css?family=Roboto");
/** Use a SCSS loop to create the transition classes to be used
by the react-transition-group add-on */
.home-transition-0-appear {
  opacity: 0.01; }

.home-transition-0-appear.home-transition-0-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 0s; }

.home-transition-1-appear {
  opacity: 0.01; }

.home-transition-1-appear.home-transition-1-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 2s; }

.home-transition-2-appear {
  opacity: 0.01; }

.home-transition-2-appear.home-transition-2-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 4s; }

.home-transition-3-appear {
  opacity: 0.01; }

.home-transition-3-appear.home-transition-3-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 6s; }

.home-transition-4-appear {
  opacity: 0.01; }

.home-transition-4-appear.home-transition-4-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 8s; }

.home-transition-5-appear {
  opacity: 0.01; }

.home-transition-5-appear.home-transition-5-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in 10s; }

.home-text {
  font-size: 16px;
  font-family: "Source Code Pro", monospace, sans-serif; }
  @media screen and (min-width: 480px) {
    .home-text {
      font-size: 18px; } }
  @media screen and (min-width: 768px) {
    .home-text {
      font-size: 22px; } }
  @media screen and (min-width: 1024px) {
    .home-text {
      font-size: 28px; } }
  .home-text--large {
    font-size: 16px; }
    @media screen and (min-width: 480px) {
      .home-text--large {
        font-size: 22px; } }
    @media screen and (min-width: 768px) {
      .home-text--large {
        font-size: 28px; } }
    @media screen and (min-width: 1024px) {
      .home-text--large {
        font-size: 36px; } }
