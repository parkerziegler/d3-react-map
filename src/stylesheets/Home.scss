@import url('https://fonts.googleapis.com/css?family=Source+Code+Pro');
@import './_constants.scss';

$sourcecode: 'Source Code Pro', monospace;
$home-font-sizes: (
  null: 16px,
  mobile: 18px,
  tablet: 22px,
  desktop: 28px
);

$large-home-font-sizes: (
  null: 16px,
  mobile: 22px,
  tablet: 28px,
  desktop: 36px
);

/** Use a SCSS loop to create the transition classes to be used
by the react-transition-group add-on */
@for $i from 0 through 5 {
  .home-transition-#{$i}-appear {
    opacity: 0.01;
  }

  .home-transition-#{$i}-appear.home-transition-#{$i}-appear-active {
    opacity: 1;
    transition: opacity 1s ease-in $i * 2s;
  }
}

.home-text {
  @include font-size($home-font-sizes);
  font-family: $sourcecode, sans-serif;
}

.home-text-large {
  @include font-size($large-home-font-sizes);
  font-family: $sourcecode, sans-serif;
}
