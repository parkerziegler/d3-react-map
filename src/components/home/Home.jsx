import * as React from 'react';
import { connect } from 'react-redux';
import '../../App.css';
import './Home.css';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import moment from 'moment';

class Home extends React.Component {

    render() {

        const { maps } = this.props;
        
        // first calculate how many shootings occurred per day on average
        let totalCount = Number(maps.shootingsData.length).toLocaleString();
        
        // use moment to get the total number of days in the dataset
        let numDays = moment("2016-12-31").diff(moment("2015-01-01"), "days");

        // get the average number of shootings per data
        let numPerDay = Number(maps.shootingsData.length / numDays).toFixed(1);

        return (
                <div className='page-content' style={{flexDirection: 'column'}}>
                    <CSSTransitionGroup
                    transitionName="header-transition"
                    transitionAppear={true}
                    transitionAppearTimeout={0}
                    transitionEnter={false}
                    transitionLeave={false}>
                        <div className='home-text'>Between January 1, 2015 and December 31, 2016,</div>
                    </CSSTransitionGroup>
                    <CSSTransitionGroup
                    transitionName="header-transition"
                    transitionAppear={true}
                    transitionAppearTimeout={2000}
                    transitionEnter={false}
                    transitionLeave={false}>
                        <div className='home-text' style={{fontSize: 36}}><b>{totalCount}</b></div>
                    </CSSTransitionGroup>
                    <CSSTransitionGroup
                    transitionName="header-transition"
                    transitionAppear={true}
                    transitionAppearTimeout={3500}
                    transitionEnter={false}
                    transitionLeave={false}>
                        <div className='home-text'>people were killed by police in the United States.</div>
                    </CSSTransitionGroup>
                    <CSSTransitionGroup
                    transitionName="header-transition"
                    transitionAppear={true}
                    transitionAppearTimeout={5000}
                    transitionEnter={false}
                    transitionLeave={false}>
                        <div className='home-text'>That amounts to roughly <b>{numPerDay}</b> people per day.</div>
                    </CSSTransitionGroup>
                </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
  return {
    maps: state.mapReducer
  };
};

export default connect(mapStateToProps)(Home);
