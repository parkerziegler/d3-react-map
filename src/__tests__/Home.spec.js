import { store } from '../index';
import Home from '../components/home/Home';

describe('<Home />', () => {
  it('renders without crashing', () => {
    shallow(<Home store={store} />);
  });

  it('matches the Jest snapshot', () => {
    const wrapper = mount(<Home store={store} />);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});
