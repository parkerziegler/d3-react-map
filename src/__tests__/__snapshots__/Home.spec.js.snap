// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the Jest snapshot 1`] = `"<div class=\\"page-content\\" style=\\"flex-direction: column;\\"><div class=\\"home-text home-transition-0-appear home-transition-0-appear-active\\">Between January 1, 2015 and December 31, 2016</div><div class=\\"home-text home-text--large home-transition-1-appear home-transition-1-appear-active\\"><b>0</b> people</div><div class=\\"home-text home-transition-2-appear home-transition-2-appear-active\\">were killed by police in the United States.</div><div class=\\"home-text home-transition-3-appear home-transition-3-appear-active\\">That amounts to roughly <b class=\\"home-text-large\\">0.0</b> people per day.</div><div class=\\"home-text home-transition-4-appear home-transition-4-appear-active\\">Here's what we know.</div><div class=\\"home-text home-transition-5-appear home-transition-5-appear-active\\" style=\\"font-family: HelveticaNeue, Helvetica, sans-serif; margin-top: 5%; font-weight: 900;\\">An Investigation | By Parker Ziegler</div></div>"`;
